drivers/serial/Kconfig.gecko

```diff
@@ -10,6 +10,6 @@ config UART_GECKO
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
 	select SOC_GECKO_USART
-	select PINCTRL if SOC_FAMILY_SILABS_S1
+	select PINCTRL
```

Comment:
This change should be in a separate commit, since it's not directly related to the board addition (may still be in the same PR, though)